main.py

```diff
@@ -183,167 +190,120 @@ def compose_message(issues):
-        return
-    # split message into command and arguments
-    query = data.strip().split(' ')
-    command = query[0][1:]
-    args = query[1:]
-
+def modify_params(ctx, conf, name, number_of_issues, noun=None):
+    error_reply = discord.Embed(color=discord.Color.purple())
+    author=None
+
+    if number_of_issues < 1:
+        error_reply.description = "I can't give you un-natural issues." + \
+            " I'm not a monster!!"      
+    if number_of_issues > 10:
+        number_of_issues = 10
+        error_reply.description = "Let's curb that enthusiasm.. just a little"
+    
+    if noun:
+        if name=="review":
+            if noun not in ["all", "ALL"]:
+                author = ctx.message.author
+                if isinstance(author, discord.Member):
+                    author = author.nick or author.name
+                elif isinstance(author, discord.User):
+                    author = author.name
+        elif noun in ["issue", "issues", "ISSUE", "ISSUES"]:
+            conf["pull_request"]=False
+        elif noun in ["pr", "prs", "PR", "PRs", "PRS"]:
+            conf["pull_request"]=True
+    # else, default to issue and no author
+    else:
+        conf["pull_request"]=False
+        author=None
+
+    error_handler(ctx.channel)
+
+    return number_of_issues, conf, author, error_reply
+
+
+for name, conf in command_config.items():
+    @bot.command(name=name)
+    async def magic_function(ctx, number_of_issues: int, noun, name=name, conf=conf): # name and conf not needed, used to prevent late binding
+        number_of_issues, conf, author, error_reply=modify_params(ctx=ctx, conf=conf, name=name, number_of_issues=number_of_issues, noun=noun)
+        if error_reply.description:
+            await ctx.channel.send(embed=error_reply)
+
+        await set_playing('On The Cue')
+        async with ctx.channel.typing():
+            issues = [x async for x in get_issues(**conf)]
+            reply = discord.Embed(color=discord.Color.purple())
+
+            if name=="rand":
+                """
+                @TODO: improve algorithm to save queries
+                1. get first batch of github_max (100), find total number
+                2. if total < requested, return all
+                3. Generate requested random numbers
+                4. get issues till max(generated_list)
+                5. return them
+                """
+                chosen_issues = random.choices(issues, k=number_of_issues)
+                reply.title = f"{number_of_issues} random picks out of {len(issues)}:"
+
+            elif name in ["review", "feedback"]:
+                if author:
+                    issues = pr_with_pending_review(get_pr_details(issues), author)
+                    # since async islice doesn't exist
+                    async for issue in issues:
+                        chosen_issues.append(issue)
+                        if len(chosen_issues) == number_of_issues:
+                            break
```

Comment:
still troubling.

Reply:
What's the issue exactly?

Reply:
Everything works, except for personal review. For me, it should generate `There weren't enough`. It fetches issues but the control doesn't go in this loop, due to which `chosen_issues` isn't updated.
The code is used as `async islice`, `issues` is an AsyncGenerator.

In the original code, it works fine, which is weird because it should work here too.

Reply:
1. `except for personal review`: Personal feedback is working?
2. Is control reaching L261?
3. Is `author == None`?

Reply:
1. No such thing as personal feedback, sir. :laughing: 
2. Nope, that's the issue
3. Nope, if author==None, i.e for general review (`all`) and feedback case, it works fine (gives prs/issues for the number specified). L267 is executed.

Reply:
Check if the generator is empty.

Reply:
We need to access it via `async for` loop, which isn't working. Is there any other method?

Reply:
Hop over to discord